src/usage_metrics/raw/s3.py

```diff
@@ -0,0 +1,61 @@
+"""Extract data from S3 logs."""
+
+from pathlib import Path
+
+import pandas as pd
+from dagster import (
+    AssetExecutionContext,
+    WeeklyPartitionsDefinition,
+    asset,
+)
+from google.cloud import storage
+from tqdm import tqdm
+
+BUCKET_URI = "pudl-s3-logs.catalyst.coop"
+LOCAL_DIR = "data/pudl_s3_logs/"
+# if not Path.exists(Path(LOCAL_DIR)):
+#     Path.mkdir(LOCAL_DIR)
+
+
+def download_s3_logs_from_gcs(
+    partition_dates: tuple[str],
+) -> list[Path]:
+    """Download all logs from GCS bucket.
+
+    If the file already exists locally don't download it.
+    """
+    bucket = storage.Client().get_bucket(BUCKET_URI)
+    blobs = bucket.list_blobs()
+    blobs = [blob for blob in blobs if blob.name.startswith(partition_dates)]
+    file_paths = []
+    for blob in tqdm(blobs):
+        path_to_file = Path(LOCAL_DIR, blob.name)
```

Comment:
We'll need to make sure that `LOCAL_DIR` exists in all our execution environments.

Looks like this is just a temporary place to store the files before we shove them into dataframes - could this just be a `NamedTemporaryDirectory`? We could pass in the `target_directory` as an argument to this function so that the lifecycle management of the tmpdir can be done in the caller.

I guess the tmpdir approach would lose the caching functionality. We could read the LOCAL_DIR from an env var and then default to tmpdir, to allow for caching locally but not have to fuss with the GH runner.